### DISCUSSION TITLE
characters not showing correctly

### DISCUSSION THREAD

**Discussion by unluckychild** [artkonekt/pdf-invoice#54] [Q&A]
So i have been looking into this problem and it is on pfdf side, since they dont support utf8. What im trying to do is make the script support vietnamese. So far i could make pdf generate vietnamese, but cant integrate lite version of pfdf into this, can someone update this pls?
Here's updated version and imo better than current version of pfdf

http://www.fpdf.org/en/script/script92.php

**Comment by fulopattila122** [comment]
Could you please specify what exactly would you like to be updated?

**Reply by unluckychild** [reply]
current invoice script is using PDFD library. But FPDF doesnt support UTF8. The link above is modified and verified version of FPDF, which supports ut8. I think it is better to use that library intead of original fpdf

**Comment by ts1985** [comment]
Would help also me and the other open discussions. The problem is the encoding which doesn't work with the current used version of FPDF. So currently it's not possible to create invoices with encodings like cyrillic etc.

**Comment by fulopattila122** [comment]
So switching over to this package https://packagist.org/packages/setasign/tfpdf would fix the problem, right?

**Reply by ts1985** [reply]
I have not tested it. But if I understand it right, this will add UTF8 Support. So with this and adding a font which has the right symbols for the language (Russian, Chinese whatever), it should be able to display the invoices also in such languages.

**Reply by unluckychild** [reply]
yes, thats correct

**Comment by unluckychild** [comment]
OK, so i tried to update the fpdf with tfpdf version and the tfpdf version WORKS with utf8 characters. But then i used your script to generate and it doesnt support utf8. So problem after updating fpdf is in the code generating invoice
Heres my tested code with utf chars

include '../src/InvoicePrinter.php';
$invoice = new InvoicePrinter();
$invoice->AddPage();
$invoice->AddFont('DejaVu','','DejaVuSansCondensed.ttf',true);
$invoice->SetFont('DejaVu','',14);
$invoice->Write(5,'Xin chào tôi là người việt nam');
$invoice->Output();

**Reply by unluckychild** [reply]
Alright. new progress. Problem lies in the font. In your script, you have public $font = 'helvetica'; 
your helvetica doesnt support utf8. and I seem to cant add new font into your script ()

// Add a Unicode font (uses UTF-8)
$invoice->AddFont('DejaVu','','DejaVuSansCondensed.ttf',true);
$invoice->SetFont('DejaVu','',14);

**Reply by unluckychild** [reply]
New progression:
I added vietnamese characters directly into fpdf in function cell, which is the most used in your case  AND function write

$invoice->Write(5,'Xin chào tôi là người việt nam');

The text directly from fpdf doesnt show correctly
The text from Write shows correctly. Both are on same page

**Reply by unluckychild** [reply]
alright. In the end i switched to mpdf and used my own HTML+CSS invoice. Easier and faster
